### ISSUE TITLE
The last character is duplicated with the VIM editor being used through xterm.js

### ISSUE THREAD

**Issue by thecrazzymouse** [xtermjs/xterm.js#3732]
## Details
The last character in a line is duplicated by VIM to signal line wrapping. This is implanted in VIM in 
https://github.com/vim/vim/blob/v8.0.0275/src/screen.c on line 5720.

Two issues have been filed for VIM, 
https://github.com/vim/vim/issues/6022
https://github.com/vim/vim/issues/1626

but actually this a xterm.js issue not handling the line wrapping as xterm.

### Steps to reproduce

1. Open file with vim that has a line longer then 20 characters
2. Execute :set columns:19

The last characters will appear twice (signalling line wrapping)

For example if operations is wrapped between the t and I character then vim will send  the operatt\u001b[17;1Hions
sequence.

**Comment by jerch** [comment]
The duplicated char is meant to be written at the right side off the screen to indicate a soft line-wrap. Not sure where you have tested it, but you see this duplicate prolly due to the fact, that your xterm.js integration does not implement CSI 8 t, thus will not resize the terminal at all, means the duplicate is just printed normally to the still bigger buffer. At least the demo does not implement CSI 8 t. Most of CSI t is left to integrators, either due to security concerns or layout/design side effects, that hinder blueprinting from the terminal core lib.

Long story short - this work as intended as far as xterm.js core is involved. Maybe ask your integration vendor to fix its CSI t implementation.

**Comment by thecrazzymouse** [comment]
Even if the terminal window is resized according to the CSI 18; command that Reports terminal size in characters.
            Response: CSI 8 ; y ; x t
              y    Terminal height in characters. (Lines)
              x    Terminal width in characters. (Columns)
when the content is copied from the terminal that still holds duplicate characters. How should we know from the sent operatt\u001b[17;1Hions sequence that the last two t characters are duplicates to signal wrapping and is just a normal duplicate letter?

**Comment by jerch** [comment]
Idk if I understand your question - who is "we", in which context?

The idea of overprinting to the right side is totally bound to stateful terminal mechanics properly doing a resize on CSI 8 t. If that assumption is not met, this trick simply falls apart.

Trying to answer your question (or what I get of it) - you simply cannot in a reliable fashion, as the terminal missed the resize part of the assumption above, thus sees the duplicate as normal char in its buffer. With some heuristics you could intercept the resize command and parse it, then you'd know the new column width of the terminal, and can truncate things further to the right (the duplicates). But thats a bit like writing parts of a terminal, just to get the duplicates right. Easier is to fix the missing CSI 8 t support of the terminal in the first place.

Maybe rephrase your question, currently it sounds like an xy problem to me.
